spec/integrity/site_setting_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+require "i18n/duplicate_key_finder"
+
+def extract_locale(path)
+  path[/\.([^.]{2,})\.yml$/, 1]
+end
+
+PLURALIZATION_KEYS ||= ['zero', 'one', 'two', 'few', 'many', 'other']
+
+def find_pluralizations(hash, parent_key = '', pluralizations = Hash.new)
+  hash.each do |key, value|
+    if Hash === value
+      current_key = parent_key.blank? ? key : "#{parent_key}.#{key}"
+      find_pluralizations(value, current_key, pluralizations)
+    elsif PLURALIZATION_KEYS.include? key
+      pluralizations[parent_key] = hash
+    end
+  end
+
+  pluralizations
+end
+
+def is_yaml_compatible?(english, translated)
+  english.each do |k, v|
+    if translated.has_key?(k)
+      if Hash === v
+        if Hash === translated[k]
+          return false unless is_yaml_compatible?(v, translated[k])
+        end
+      else
+        return false unless v.class == translated[k].class
+      end
+    end
+  end
+
+  true
+end
+
+describe "site setting integrity checks" do
+  let(:site_setting_file) { File.join(Rails.root, 'config', 'site_settings.yml') }
+  let(:yaml) { YAML.load_file(site_setting_file) }
+
+  it "does not contain invalid interpolation keys" do
+    matches = File.read(site_setting_file).scan(/%\{([^a-zA-Z\s]+)\}|\{\{([^a-zA-Z\s]+)\}\}/)
```

Comment:
Hmm mayeb try to use `I18nInterpolationKeysFinder.find` here or `I18n::INTERPOLATION_PATTERN`